trainer.py

```diff
@@ -112,9 +111,8 @@ def evaluate_model_on_dataloader(
                 loss = self.criterion(predictions, targets)
 
                 correct_labeled_samples += (predictions.argmax(1) == targets).type(torch.int).sum().item()
-                nof_samples += len(inputs)
                 total_loss += loss.item()
-                avg_loss = total_loss / nof_samples
+                avg_loss = total_loss / (batch_idx + 1)
                 accuracy = correct_labeled_samples * 100 / nof_samples
```

Comment:
should be:
accuracy = correct_labeled_samples * 100 / ((batch_idx + 1) * self.batch_size))

you didnt go over all the samples yet so its unfair to divide by the entire set